downstairs/src/lib.rs

```diff
@@ -3283,22 +3230,16 @@ impl Work {
         self.active.get(&ds_id).unwrap().clone()
     }
 
-    /**
-     * If the requested job is still new, and the dependencies are all met,
-     * return the job ID and the upstairs UUID, moving the state of the job as
-     * InProgress. If the dependencies are not met, move the state to DepWait.
-     *
-     * If this job is not new, then just return None.
-     *
-     * If the job is InProgress, return itself.
-     */
-    fn in_progress(&mut self, ds_id: JobId) -> Option<JobId> {
-        /*
-         * Once we support multiple threads, we can obtain a ds_id that
-         * looked valid when we made a list of jobs, but something
-         * else moved that job along and now it no longer exists.  We
-         * need to handle that case correctly.
-         */
+    /// If the requested job is still new, and the dependencies are all met,
+    /// move the state of the job to `InProgress` and return `true`.
+    ///
+    /// If the dependencies are not met, move the state to `DepWait` and return
+    /// `false`.
+    ///
+    /// If this job is not new, then just return `false`.
+    ///
+    /// If the job already is `InProgress`, return `true` (to be idempotent).
```

Comment:
We should move this comment above the "not new, return false" comment, so there is no
confusion on what the flow through here is.

And, while we are at it, we should change the "not new" to be `Done`, as that is the only
other choice besides `InProgress`, `New`, or `DepWait`.

Reply:
Good idea, fixed